### PR TITLE
fix: min timeout should be DEFAULT_CONNECTION_TIMEOUT + 50

### PR DESCRIPTION
**Pull request by OrrLumigo**

### CODE REVIEW

File: src/utils.test.js [hunk 1]
```diff
@@ -490,10 +490,10 @@ describe('utils', () => {
     expect(utils.getTracerMaxDurationTimeout()).toEqual(500);
   });
 
-  test('getTracerMaxDurationTimeout -> 20% of the run time', () => {
+  test('getTracerMaxDurationTimeout -> max(20%,350) of the run time', () => {
     const inputs = new HandlerInputesBuilder().withTimeout(1000).build();
     TracerGlobals.setHandlerInputs(inputs);
-    expect(utils.getTracerMaxDurationTimeout()).toEqual(200);
+    expect(utils.getTracerMaxDurationTimeout()).toEqual(350);
```
Comment (doriaviram) [inline src/utils.test.js:496]: Change the timeout of the function to check the 20% logic

File: src/utils.ts [hunk 2]
```diff
@@ -199,8 +199,7 @@ export const getTracerMaxDurationTimeout = (): number => {
   if (process.env[TRACER_TIMEOUT_FLAG]) return parseFloat(process.env[TRACER_TIMEOUT_FLAG]);
   const { context } = TracerGlobals.getHandlerInputs();
   if (isAwsContext(context)) {
-    const { remainingTimeInMillis } = getContextInfo(context);
-    return Math.min(DEFAULT_TRACER_MAX_DURATION_TIMEOUT, remainingTimeInMillis / 5);
```
Comment (doriaviram) [inline src/utils.ts:203]: Its looks like you deleted this logic, what about small timeout functions?

**Review by doriaviram — CHANGES_REQUESTED** [review]

File: src/globals.ts [hunk 3]
```diff
@@ -118,6 +118,9 @@ export const TracerGlobals = (() => {
     event: {},
     context: {},
   };
+
+  let lambdaTimeout = DEFAULT_TRACER_TIMEOUT;
```
Comment (saartochner) [inline src/globals.ts:122]: maybe "duration" is a better name?

Comment (OrrLumigo) [reply src/globals.ts]: left or right side ?

File: src/globals.ts [hunk 4]
```diff
@@ -118,16 +119,30 @@ export const TracerGlobals = (() => {
...
     isStepFunction: false,
     maxSizeForRequest: DEFAULT_MAX_SIZE_FOR_REQUEST,
+    lambdaTimeout: MAX_TRACER_ADDED_DURATION_ALLOWED,
+  };
+
+  const setHandlerInputs = ({ event, context }) => {
+    Object.assign(tracerInputs, {
+      lambdaTimeout: context?.getRemainingTimeInMillis
+        ? context.getRemainingTimeInMillis()
+        : MAX_TRACER_ADDED_DURATION_ALLOWED,
```
Comment (doriaviram) [inline src/globals.ts:137]: leave this as undefined, let the user choose how to handle

**Pull request merged**